deps/wazuh_testing/wazuh_testing/system/host_manager.py

```diff
@@ -0,0 +1,472 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import tempfile
+import testinfra
+import base64
+import os
+from ansible.inventory.manager import InventoryManager
+from ansible.parsing.dataloader import DataLoader
+from ansible.vars.manager import VariableManager
+
+
+class HostManager:
+    """Remote host management interface.
+
+    It allows to manage remote hosts using ansible inventory and testinfra framework.
+    """
+
+    def __init__(self, inventory_path):
+        """Constructor of host manager class.
+
+        Args:
+            inventory_path (str): Ansible inventory path
+        """
+        self.inventory_path = inventory_path
+
+        data_loader = DataLoader()
+        self.inventory_manager = InventoryManager(loader=data_loader, sources=inventory_path)
+        self.variable_manager = VariableManager(loader=data_loader, inventory=self.inventory_manager)
+
+    def get_host(self, host):
+        """Get the testinfra host.
+
+        Args:
+            host (str): Hostname
+        Returns:
+            testinfra.modules.base.Ansible: Host instance from hostspec
+        """
+        return testinfra.get_host(f"ansible://{host}?ansible_inventory={self.inventory_path}")
+
+    def get_groups(self):
+        """Get the groups of the inventory.
+
+        Returns:
+            list: Groups of the inventory
+        """
+        return list(self.inventory_manager.groups.keys())
+
+    def get_group_hosts(self, pattern=None):
+        """Get all hosts from inventory that belong to a group.
+
+        Args:
+            group (str): Group name
+        Returns:
+            list: List of hosts
+        """
+        return self.inventory_manager.get_hosts(pattern=pattern) if pattern else self.inventory_manager.get_hosts()
+
+    def get_host_variables(self, host):
+        """Get the Ansible object for communicating with the specified host.
+
+        Args:
+            host (str): Hostname
+        Returns:
+            testinfra.modules.base.Ansible: Host instance from hostspec
+        """
+        inventory_manager_host = self.inventory_manager.get_host(host)
+        return self.variable_manager.get_vars(host=inventory_manager_host)
+
+    def collect_host_ansible_facts(self, host):
+        """Get the ansible facts of the specified host.
+
+        Args:
+            host (str): Hostname
+        Returns:
+            str: OS of the host
+        """
+        testinfra_host = self.get_host(host)
```

Comment:
```suggestion
        testinfra_host = self.get_host(host)
        
```

Reply:
Separate the `return` sentence if there are previous lines of implementation. Apply the same to the other occurrences.

Reply:
Done e04ee4b